aten/src/ATen/functorch/BatchRulesHelper.cpp

```diff
@@ -160,20 +160,9 @@ void vmapIncompatibleInplaceError(const char* schema_name) {
     "please file a bug report instead.");
 }
 
-static void handleScalarTypePromotion(Tensor& logical_scalar_tensor, Tensor& second) {
-  auto result_type = at::native::result_type(logical_scalar_tensor[0], second);
-  if (logical_scalar_tensor.scalar_type() != result_type) {
-    logical_scalar_tensor = logical_scalar_tensor.to(result_type);
-  }
-  if (second.scalar_type() != result_type) {
-    second = second.to(result_type);
-  }
```

Comment:
The problem I saw was that these `.to` calls on tensors were being traced as `_to_copy` on scalars. I don't see anything wrong with the `.to` calls themself, I think it's something specific to this being called when tracing.

Reply:
Yeah I don't understand how the scalar could be treated as a Tensor in this. Somehow when tracing we lose track of the fact that these are scalars.